internal/bminventory/inventory.go

```diff
@@ -1315,11 +1316,73 @@ func (b *bareMetalInventory) TransformClusterToDay2Internal(ctx context.Context,
 		return nil, common.NewApiError(http.StatusInternalServerError, err)
 	}
 
+	log.Infof("Pre-update TransformClusterToDay2Internal APIVIP %v", cluster.APIVip)
+	log.Infof("Pre-update TransformClusterToDay2Internal APIVIPDNSName %v", cluster.APIVipDNSName)
+	if cluster.IgnitionEndpoint == nil || cluster.IgnitionEndpoint.URL == nil {
```

Comment:
Maybe i missed something but why it was moved here? Currently it will kill day2 on podman installations.
Moving to https should be relevant only in ztp flow.

Reply:
I think that by ZTP you mean kube-api.
To my understanding fetching the day 2 worker via HTTPS requires CA cert.
current me if I'm wrong but this is how I understand the "if possible", if we can get the CA cert we can/should use HTTPS.
We can get the cert only when installing with kube-api (only in kubeapi we have access to the spoke cluster).

Reply:
@romfreiman @eranco74 @avishayt IMO better to update crd and with new endpoint and in reconciliation should update cluster with relevant data. I don't think that transform cluster should do those changes as behavior is different for different setups. WDYT?

Reply:
Why not move this logic to the controller?